From 17830918f1bebf3696c6660f602de84d3efeb1ac Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Thu, 17 Apr 2014 15:50:44 +0200
Subject: [PATCH] Fedora hack: Make the suid-root wrapper always start the
 server with root rights

Do NOT upstream.

Since most display managers are not ready yet to start Xorg in way which will
keep it working without root-rights, see:
https://fedoraproject.org/wiki/Changes/XorgWithoutRootRights

Just keep starting X as root for now, but do it through the wrapper. This
allows people who want to test Xorg without root rights to create a
/etc/X11/Xwrapper.config file with:

needs_root_rights = auto

In there to test XorgWithoutRootRights without breaking people's setups due
to non ready display-managers.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 hw/xfree86/xorg-wrapper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/xfree86/xorg-wrapper.c b/hw/xfree86/xorg-wrapper.c
index 90c8c11..539bfe4 100644
--- a/hw/xfree86/xorg-wrapper.c
+++ b/hw/xfree86/xorg-wrapper.c
@@ -153,7 +153,7 @@ int main(int argc, char *argv[])
     int kms_cards = 0;
     int total_cards = 0;
     int allowed = CONSOLE_ONLY;
-    int needs_root_rights = -1;
+    int needs_root_rights = 1;
 
     progname = argv[0];
 
-- 
1.9.0

